python-impl/keys.py

```diff
@@ -83,6 +87,31 @@ def from_bytes(buffer):
     def from_seed(seed):
         hashed = hmac256(seed, b"BLS private key seed")
         return PrivateKey(int.from_bytes(hashed, "big") % default_ec.n)
+    
+    @staticmethod
+    def new_threshold(T, N):
+        """
+        Create a new private key with associated data suitable for
+        T of N threshold signatures under a Joint-Feldman scheme.
+        
+        After the dealing phase, one needs cooperation of T players
+        out of N in order to sign a message with the master key pair.
+
+        Return:
+          - poly[0] - your share of the master secret key
+          - commitments to your polynomial P
+          - secret_fragments[j] = P(j), to be sent to player j
+            (All N secret_fragments[j] can be combined to make a secret share.)
+        """
+        assert 1 <= T <= N
+        g1 = generator_Fq()
+        poly = [Fq(default_ec.n, RNG.randint(1, default_ec.n - 1))
+                for _ in range(T)]
+        commitments = [g1 * c for c in poly]
+        secret_fragments = [sum(c * pow(x, i, default_ec.n) for i, c in enumerate(poly))
```

Comment:
So, T players here can cooperate to find all of the polynomial coefficients correct?

Reply:
If that's the case, we should warn people to never use their existing keys as coefficients/commitments

Reply:
Not sure what you mean: the polynomial coefficients are randomly chosen.